src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugRpcModule.cs

```diff
@@ -65,6 +70,23 @@ public ResultWrapper<int> debug_deleteChainSlice(in long startNumber, bool force
 
     public ResultWrapper<GethLikeTxTrace> debug_traceTransaction(Hash256 transactionHash, GethTraceOptions? options = null)
     {
+        Hash256? blockHash = _debugBridge.GetTransactionBlockHash(transactionHash);
+        if (blockHash is null)
+        {
+            return ResultWrapper<GethLikeTxTrace>.Fail($"Cannot find block hash for transaction {transactionHash}", ErrorCodes.ResourceNotFound);
+        }
+
+        SearchResult<BlockHeader> searchResult = _blockFinder.SearchForHeader(new BlockParameter(blockHash));
```

Comment:
`FindHeader` is simpler

Reply:
Done in [ecd0e0b](https://github.com/NethermindEth/nethermind/pull/8621/commits/ecd0e0b49e85023671b06d146423808d82537264)